crates/polars-pipe/src/executors/sinks/ordered.rs

```diff
@@ -53,8 +54,23 @@ impl Sink for OrderedSink {
             )));
         }
         self.sort();
+
         let chunks = std::mem::take(&mut self.chunks);
-        Ok(FinalizedSink::Finished(chunks_to_df_unchecked(chunks)))
+        let mut combiner = SemicontiguousVstacker::default();
```

Comment:
I think a `rechunk_par` will be faster here. `vstack mut` will lead to reallocations on a quadratic strategy. Where a `rechunk` reallocates and knows how much to allocate up front. After that all can be memcopied.

I still think we shouldn't `rechunk` by default as now we always pay that cost even if you just want to see the results.

Most operations that come after will know if they need a rechunk or not to work optimal. So we shouldn't that decide for the consumer of this node.

Reply:
1. I will try `rechunk_par` tomorrow.
2. I ... am not sure it's currently quadratic since there's only one rechunking per 4MB block, in `SemicontiguousVstacker::flush()`? But my mental model of `vstack_mut` may be wrong, but I thought it was just pointer copying until the point where you actually rechunk?
3. `write_parquet()` is definitely _not_ rechunking, thus the problem. So another option is moving the rechunking there instead of here.

Reply:
1. I assume you mean `as_single_chunk_par()`?
2. I remembered why I'm not doing this: the goal is _not_ to do a single chunk, cause that means a huge allocation, doubling memory usage. Whereas doing e.g. 4MB at a time means you effectively have the same memory usage, while still getting the benefit of having none-tiny chunks.

To rephrase: this PR is a half-way point between "stick to a massive number of tiny chunks" and "rechunk into a single chunk", where it tries to have no real significant increase in memory requirements, and hopefully not too much of a performance hit.